lib/launchers/ovirt.rb

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env ruby
+require 'ovirtsdk4'
+
+require 'common'
+require 'host'
+require 'launchers/cloud_helper'
+
+module BushSlicer
+
+  class Ovirt
+    include Common::Helper
+    include Common::CloudHelper
+
+    attr_reader :config
+    attr_reader :trusted_ca_path
+    attr_reader :ovirt_creds
+
+    def initialize(**opts)
+      @can_terminate = true
+      service_name = opts[:service_name]
+      if service_name
+        @config = conf[:services, service_name]
+      else
+        @config ||= :OVIRT
+      end
+      config_opts = opts[:config]
+      if config_opts
+        deep_merge!(@config, config_opts)
+      end
+
+      @pem_url = @config[:api_url].gsub(/\/api\/?$/,"/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA")
+
+      ca_path = @config[:ovirt_ca_file]
+      @trusted_ca_path = trust_ovirt_certificate(ca_path)
+      @ovirt_creds = generate_ovirt_creds(trusted_ca_path)
+      @connection = OvirtSDK4::Connection.new(
+          url: @config[:api_url],
+          username: @config[:userdomain],
+          password: @config[:password],
+          insecure: @config[:insecure],
+          ca_file:  @trusted_ca_path
+      )
+    end
+
+    def get_resource(service, query=nil)
+      svc = @connection.system_service.send("#{service}_service")
+      if query
+        return svc.list(search: query)
+      else
+        return svc.list
+      end
+    end
+
+    def get_cluster_id(name)
+      cluster = get_resource(service='clusters', query="name=#{name}")[0]
```

Comment:
Guilherme De Oliveira Santos, do you think we need to check if the `cluster` returned is a valid object?...in order words, you are alway getting the first element of the `get_resource` return value...can it be empty?   if so then call to `cluster.id` will probably cause issue.

Reply:
I see @pruan-rht , in fact `get_resource` should return a list o elements and indeed the list can be empty (It will always return a list as long as it not fails). In this case (empty list), do you have any specific suggestion on how to handle it? I'm thinking on raise an error with a proper message

Reply:
@santos1709 probably check for the size of the return list to see if it's empty first instead to make sure it's not `nil`..otherwise you will get ```NoMethodError: undefined method `id' for nil:NilClass``` when you do cluster.id.   Whether or not to raise an exception depends on your use case.

Reply:
Okay, it should always return something or it will break ahead.

Reply:
@pruan-rht Changes done, once you have some time, PTAL, thanks!

Reply:
I see the job here fails when trying to install ovirt-engine-sdk gem. This gem needs the following requirements: `gcc`,
`libcurl-devel`, `libxml2-devel`. Maybe the job runner doesn't have one of them? @pruan-rht